osu.Game/Screens/Play/ReplayPlayer.cs

```diff
@@ -1,42 +1,46 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
+using System.Collections.Generic;
 using System.Threading.Tasks;
 using osu.Framework.Input.Bindings;
+using osu.Game.Beatmaps;
 using osu.Game.Input.Bindings;
+using osu.Game.Rulesets.Mods;
 using osu.Game.Scoring;
 using osu.Game.Screens.Ranking;
 
 namespace osu.Game.Screens.Play
 {
     public class ReplayPlayer : Player, IKeyBindingHandler<GlobalAction>
     {
-        protected readonly Score Score;
+        private readonly Func<IBeatmap, IReadOnlyList<Mod>, Score> createScore;
 
         // Disallow replays from failing. (see https://github.com/ppy/osu/issues/6108)
         protected override bool CheckModsAllowFailure() => false;
 
         public ReplayPlayer(Score score, PlayerConfiguration configuration = null)
-            : base(configuration)
+            : this((_, __) => score, configuration)
         {
-            Score = score;
         }
 
-        protected override void PrepareReplay()
+        public ReplayPlayer(Func<IBeatmap, IReadOnlyList<Mod>, Score> createScore, PlayerConfiguration configuration = null)
+            : base(configuration)
         {
-            DrawableRuleset?.SetReplayScore(Score);
+            this.createScore = createScore;
         }
 
-        protected override Score CreateScore()
+        protected override void PrepareReplay()
         {
-            var baseScore = base.CreateScore();
-
-            // Since the replay score doesn't contain statistics, we'll pass them through here.
-            Score.ScoreInfo.HitEvents = baseScore.ScoreInfo.HitEvents;
+            DrawableRuleset?.SetReplayScore(Score);
 
-            return Score;
+            // todo: move to base class along with Score?
+            ScoreProcessor.NewJudgement += result => ScoreProcessor.PopulateScore(Score.ScoreInfo);
```

Comment:
I think this isn't required anymore - isn't this already in the base?

Same as `SpectatorPlayer()` - it also overrides `PrepareReplay()` in a similar fashion.